File: LegacySource/settlement/room/infra/export/ExportTally.cs
using System;
using System.IO;
using settlement.room.infra.export;
using init.resources;
using settlement.main;
using snake2d.util.file;
using util.data;
using util.keymap;

public sealed class ExportTally
{
    private readonly RMapInt<RESOURCE> pAmount = new RMapInt<RESOURCE>(RESOURCES.map());
    private readonly RMapInt<RESOURCE> pCapacity = new RMapInt<RESOURCE>(RESOURCES.map());
    public readonly INT_O<RESOURCE> amount = pAmount;
    public readonly INT_O<RESOURCE> capacity = pCapacity;

    public ExportTally()
    {
    }

    private readonly SAVABLE saver = new SAVABLE
    {
        Save = (FilePutter file) =>
        {
            pAmount.Save(file);
            pCapacity.Save(file);
        },

        Load = (FileGetter file) =>
        {
            pAmount.Load(file);
            pCapacity.Load(file);
            pAmount.Clear();
            foreach (ExportInstance i in SETT.ROOMS().EXPORT.all())
            {
                if (i.Resource() != null)
                    pAmount.Inc(i.Resource(), i.Amount());
            }
        },

        Clear = () =>
        {
            pAmount.Clear();
            pCapacity.Clear();
        }
    };

    public void Inc(RESOURCE r, int amount, int capacity)
    {
        this.pAmount.Inc(r, amount);
        this.pCapacity.Inc(r, capacity);
    }
}

File: settlement/room/infra/export/ExportTally.java
package settlement.room.infra.export;

import java.io.IOException;

import init.resources.RESOURCE;
import init.resources.RESOURCES;
import settlement.main.SETT;
import snake2d.util.file.FileGetter;
import snake2d.util.file.FilePutter;
import snake2d.util.file.SAVABLE;
import util.data.INT_O;
import util.keymap.RMapInt;

public final class ExportTally {
	

	private final RMapInt<RESOURCE> pAmount = new RMapInt<RESOURCE>(RESOURCES.map());
	private final RMapInt<RESOURCE> pCapacity =new RMapInt<RESOURCE>(RESOURCES.map());
	public final INT_O<RESOURCE> amount = pAmount;
	public final INT_O<RESOURCE> capacity = pCapacity;
	
	ExportTally() {

	}
	
	final SAVABLE saver = new SAVABLE() {
		@Override
		public void save(FilePutter file) {
			
			pAmount.save(file);
			pCapacity.save(file);
			
		}

		@Override
		public void load(FileGetter file) throws IOException {
			
			pAmount.load(file);
			pCapacity.load(file);
			pAmount.clear();
			for (ExportInstance i : SETT.ROOMS().EXPORT.all()) {
				if (i.resource() != null)
					pAmount.inc(i.resource(), i.amount);
			}
			
		}

		@Override
		public void clear() {
			pAmount.clear();
			pCapacity.clear();
			
		}
	};
	
	void inc(RESOURCE r, int amount, int capacity) {
		this.pAmount.inc(r, amount);
		this.pCapacity.inc(r, capacity);
	}	

}
